### PR TITLE
Increased the speed and memory efficiency for GLEM in semi-supervised/pseudolabel setting

### PR DESCRIPTION
**Pull request by wangz10**
*Issue #, if available:*

*Description of changes:*
- Rewrote the `flip_node_mask` function to increase the speed and reduce memory for getting the unlabeled node indices required for semi-supervised data loader. 
- Fixed a bug for GLEM model to get the correct `target_ntype` from config.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/dataloading/utils.py [hunk 1]
```diff
@@ -113,20 +112,21 @@ def modify_fanout_for_target_etype(g, fanout, target_etypes):
...
 
         Returns
         -------
-        flipped mask: dgl.distributed.DistTensor
+        flipped mask: torch.Tensor
     """
-    flipped_dist_tensor = deepcopy(dist_tensor)
-    for idx in range(dist_tensor.shape[0]):
-        flipped_dist_tensor[idx] = 1 - dist_tensor[idx]
+    flipped_dist_tensor = th.ones(dist_tensor.shape, dtype=dist_tensor.dtype)
```
Comment (zheng-da) [inline python/graphstorm/dataloading/utils.py:130]: this is not scalable. you should create a distributed tensor to store the flipped mask.

File: python/graphstorm/model/node_glem.py [hunk 2]
```diff
@@ -354,17 +358,17 @@ def _get_seed_nodes(self, input_nodes, blocks):
         input_nodes : {target_ntype: tensor.shape [bs], other_ntype: []}
         blocks : list[dgl.Block]
         """
-        target_ntype = blocks[-1].dsttypes[0]
+        target_ntype = self.target_ntype
```
[the remark this answers is not in the document]
Comment (wangz10) [reply python/graphstorm/model/node_glem.py:361]: Before this PR, yes.

File: python/graphstorm/dataloading/utils.py [hunk 3]
```diff
@@ -113,20 +112,27 @@ def modify_fanout_for_target_etype(g, fanout, target_etypes):
         edge_fanout_lis.append(edge_fanout_dic)
     return edge_fanout_lis
 
-def flip_node_mask(dist_tensor):
+def _init_func(shape, dtype):
+    """Initialize function for DistTensor
+    """
+    return th.ones(shape, dtype=dtype)
+
+def flip_node_mask(dist_tensor, indices):
```
Comment (zheng-da) [inline python/graphstorm/dataloading/utils.py:120]: if you want to have a general implementation, you should remove `indices`

**Comment by zheng-da** [comment]
can you add an integration test to run GLEM on a heterogeneous graph?

**Comment by wangz10** [comment]
> can you add an integration test to run GLEM on a heterogeneous graph?

The existing integration test already runs on MovieLens 100k, which is a heterogeneous graph.

**Review by zheng-da — APPROVED** [review]

**Pull request merged**